src/rest.h

```diff
@@ -14,6 +14,15 @@ enum class RESTResponseFormat {
     JSON,
 };
 
+/**
+ * Parse a URI to get the data format and URI without data format
+ * and query string.
+ *
+ * @param[out]  param       The strReq without the data format string and
+ *                          without the query string (if any).
+ * @param[in]   strReq      The URI to be parsed.
+ * @return      RESTResponseFormat   RESTResponseFormat that was parsed from the URI.
```

Comment:
nit: no reason to explicitly menton the return type here before the description (we don't do that usually as far as I saw)
```suggestion
 * @return      RESTResponseFormat that was parsed from the URI.
```